src/algebra/homology/local_cohomology.lean

```diff
@@ -0,0 +1,173 @@
+/-
+Copyright (c) 2023 Emily Witt. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Emily Witt, Scott Morrison, Jake Levinson, Sam van Gool
+-/
+import ring_theory.ideal.basic
+import algebra.category.Module.colimits
+import algebra.category.Module.projective
+import category_theory.abelian.ext
+import category_theory.limits.final
+import ring_theory.noetherian
+
+/-!
+# Local cohomology.
+
+This file defines the `i`-th local cohomology module of an `R`-module `M` with support in an
+ideal `I` of `R`, where `R` is a commutative ring, as the direct limit of Ext modules:
+
+Given a collection of ideals cofinal with the powers of `I`, consider the directed system of
+quotients of `R` by these ideals, and take the direct limit of the system induced on the `i`-th
+Ext into `M`.  One can, of course, take the collection to simply be the integral powers of `I`.
+
+## References
+
+* [M. Hochster, *Local cohomology*][hochsterunpublished]
+  <https://dept.math.lsa.umich.edu/~hochster/615W22/lcc.pdf>
+* [R. Hartshorne, *Local cohomology: A seminar given by A. Grothendieck*][hartshorne61]
+* [M. Brodmann and R. Sharp, *Local cohomology: An algebraic introduction with geometric
+  applications*][brodmannsharp13]
+* [S. Iyengar, G. Leuschke, A. Leykin, Anton, C. Miller, E. Miller, A. Singh, U. Walther,
+  *Twenty-four hours of local cohomology*][iyengaretal13]
+
+## Tags
+
+local cohomology, local cohomology modules
+
+## Future work
+
+* Prove that this definition is equivalent to:
+    * the right-derived functor definition
+    * the characterization as the limit of Koszul homology
+    * the characterization as the cohomology of a Cech-like complex
+* Prove that local cohomology depends only on the radical of the ideal
+* Establish long exact sequence(s) in local cohomology
+-/
+
+open opposite
+open category_theory
+open category_theory.limits
+
+noncomputable theory
+
+universes u v
+
+/- We define local cohomology, implemented as a direct limit of Ext(R/J, --). -/
+section local_cohomology
+
+section
+variables {R : Type u} [comm_ring R] {D : Type v} [small_category D]
+
+local attribute [ext] quiver.hom.unop_inj
```

Comment:
I don't think this is needed: I commented it out and the file still compiles. Remove?